src/jaegertracing/testutils/TUDPTransport.h

```diff
@@ -62,23 +68,34 @@ class TUDPTransport
         auto clientAddrLen = static_cast<::socklen_t>(sizeof(clientAddr));
         const auto numRead =
             ::recvfrom(_socket.handle(),
+#ifdef WIN32
+                       reinterpret_cast<char*>(buf),  // this cast is safe
+#else
                        buf,
+#endif
                        len,
                        0,
                        reinterpret_cast<::sockaddr*>(&clientAddr),
                        &clientAddrLen);
         _clientAddr = net::IPAddress(clientAddr, clientAddrLen);
-        return numRead;
+        // the return value conrreponds to the size of the data read from the
+        // socket. upon error, recvfrom return -1. In this case, we return 0 to
+        // say that noothing was read.
+        return std::max<long>(0, numRead);
     }
 
     void write(const uint8_t* buf, uint32_t len)
     {
         ::sendto(_socket.handle(),
-                 buf,
-                 len,
-                 0,
-                 reinterpret_cast<const ::sockaddr*>(&_clientAddr.addr()),
-                 _clientAddr.addrLen());
+#ifdef WIN32
+            reinterpret_cast<const char*>(buf),  // this cast is safe
```

Comment:
That is fine with me. Any reason to make Windows specific and not on POSIX too?

Reply:
On POSIX, send/sendto expect void*. It will be weird to cast to char*. What do you think?

Reply:
Even if weird better than macros IMO.

Reply:
I personally prefer the avoiding the cast when not needed.